coralproject/talk
# [CORL-2885]: accessibility improvements in the stream

## What does this PR do?

These changes improve accessibility for the stream. They include:
* programmatically setting keyboard focus, not just visibility, when using comments links to navigate to the top of the comments and to the top of the article
* Adding `aria-expanded` attributes to elements that expand/collapse
* Ensuring that the `aria-label` for the `Go to conversation` button in featured comments includes the same text that is visibly shown

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can go to the bottom of the stream and click `Top of comments`. Hit tab. See that the programmatic focus is at the top of the comments section and no longer at the bottom of the page. Go to the bottom of the stream and click `Top of article`. Hit tab. See that the programmatic focus is the first element on the page and no longer still at the bottom of the page.

You can test that the `aria-expanded` attribute is correctly set for the ignore user settings in `My profile` --> `Preferences`. Click `Manage`/`Close` and see the attribute update on the button in dev tools. Also test comments being hidden/shown in the stream the same way when the `+`/`-` icon buttons are clicked. Also test out the `Show`/`Hide` media (such as YouTube) button in comments and see that the attribute is now set on those buttons.

Feature a comment. Inspect the comment in dev tools. See that its `aria-label` includes the same text (`Go to conversation`) as is visually shown on the `Go to conversation` button.

## Where any tests migrated to React Testing Library?



## How do we deploy this PR?
